src/Form/formatters.js

```diff
@@ -110,10 +110,145 @@ const getCountryName = (el, options = {}) => {
     return localisedCountryName
 }
 
+/**
+ * Try to get a map of localised country names to code, or falls back to the English map
+ * @param {HTMLInputElement | HTMLSelectElement} el
+ */
+const getLocalisedCountryNamesToCodes = (el) => {
+    if (typeof Intl.DisplayNames !== 'function') return COUNTRY_NAMES_TO_CODES
+
+    // Try to infer the field language or fallback to en
+    const elLocale = inferElementLocale(el)
+
+    return Object.fromEntries(
+        Object.entries(COUNTRY_CODES_TO_NAMES)
+            .map(([code]) => [getCountryDisplayName(elLocale, code), code])
+    )
+}
+
+/**
+ * Try to infer a country code from an element we identified as identities.addressCountryCode
+ * @param {HTMLInputElement | HTMLSelectElement} el
+ * @return {string}
+ */
+const inferCountryCodeFromElement = (el) => {
+    if (COUNTRY_CODES_TO_NAMES[el.value]) return el.value
+    if (COUNTRY_NAMES_TO_CODES[el.value]) return COUNTRY_NAMES_TO_CODES[el.value]
+
+    const localisedCountryNamesToCodes = getLocalisedCountryNamesToCodes(el)
+    if (localisedCountryNamesToCodes[el.value]) return localisedCountryNamesToCodes[el.value]
+
+    if (el instanceof HTMLSelectElement) {
+        const selectedText = el.selectedOptions[0]?.text
+        if (COUNTRY_CODES_TO_NAMES[selectedText]) return selectedText
+        if (COUNTRY_NAMES_TO_CODES[selectedText]) return localisedCountryNamesToCodes[selectedText]
+        if (localisedCountryNamesToCodes[selectedText]) return localisedCountryNamesToCodes[selectedText]
+    }
+    return ''
+}
+
+/**
+ * @param {InternalDataStorageObject} credentials
+ * @return {boolean}
+ */
+const shouldStoreCredentials = ({credentials}) =>
+    Boolean(credentials.password)
+
+/**
+ * @param {InternalDataStorageObject} credentials
+ * @return {boolean}
+ */
+const shouldStoreIdentities = ({identities}) =>
+    Boolean(
+        (identities.firstName || identities.fullName) &&
+        identities.addressStreet &&
+        identities.addressCity
+    )
+
+/**
+ * @param {InternalDataStorageObject} credentials
+ * @return {boolean}
+ */
+const shouldStoreCreditCards = ({creditCards}) =>
+    Boolean(creditCards.cardNumber && (
+        creditCards.cardSecurityCode ||
+        // Some forms (Amazon) don't have the cvv, so we still save if there's everything else
+        (creditCards.cardName &&
+        // Expiration can be unified or separate
+        (creditCards.expiration || (creditCards.expirationYear && creditCards.expirationMonth)))
+    ))
+
+/**
+ * Formats form data into an object to send to the device for storage
+ * If values are insufficient for a complete entry, they are discarded
+ * @param {InternalDataStorageObject} formValues
+ * @return {DataStorageObject}
+ */
+const prepareFormValuesForStorage = (formValues) => {
```

Comment:
**nit**

(again, not blocking)

Since we introduced the `InputTypes` folder, I believe that going forward we could start to use those files as a catch-all place for related logic. Meaning, if I want to see code relating to how Credentials are prepared, I'd go the `InputTypes/Credentials.js` file and expect some interface to be adhered to in there.

Reply:
We'll follow up on this.